### PR TITLE
Update documentation for getting Prometheus metrics.

### PR DESCRIPTION
**Pull request by shrinandj**
**What this PR does / why we need it**:
It updates the documentation for using KServe with Prometheus. Without this, it was super hard to get KServe metrics scraped by Prometheus.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] This is a documentation change itself.

**Feature/Issue validation/testing**:

Followed the steps mentioned in the documentation on a K8s cluster and verified that kserve metrics were getting scraped by Prometheus.

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

No changes in image versions.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
* Updated documentation for getting KServe metrics scraped by Prometheus.
```

### CODE REVIEW

**Comment by shrinandj** [comment]
Any update on this PR?

File: docs/samples/metrics-and-monitoring/README.md [hunk 1]
```diff
@@ -24,6 +24,67 @@ kustomize build docs/samples/metrics-and-monitoring/prometheus | kubectl apply -
...
+$ kubectl get prometheus -n <name-of-the-namespace> -o yaml
+...
+  serviceMonitorNamespaceSelector: {}
+  serviceMonitorSelector:
+    matchLabels:
+      release: kube-prometheus-stack-1651295153
+```
+This meant that every `ServiceMonitor` object that prometheus is expected to scrape should have the label `release: kube-prometheus-stack-1651295153`.
+
+* The ServiceMonitor objects created `knative-serving`, `knative-eventing` or even in the application namespaces should have the above label. E.g.
```
Comment (pvaneck) [inline docs/samples/metrics-and-monitoring/README.md]: s/created/created in the/

File: docs/samples/metrics-and-monitoring/README.md [hunk 2]
```diff
@@ -24,6 +24,67 @@ kustomize build docs/samples/metrics-and-monitoring/prometheus | kubectl apply -
 
 > Note: The above steps install Kubernetes resource objects in the `kfserving-monitoring` namespace. This is Kustomizable. To install under a different namespace, say `my-monitoring`, change `kfserving-monitoring` to `my-monitoring` in the following three files: a) `prometheus-operator/namespace.yaml`, b) `prometheus-operator/kustomization.yaml`, and c) `prometheus/kustomization.yaml`.
 
+* [This](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md) article provides details of how Prometheus setup using the above operator actually scrapes metrics.
```
Comment (pvaneck) [inline docs/samples/metrics-and-monitoring/README.md:29]: nit: I think would be good to have a subheader before this line like:

`### How Metrics are Scaped`

**Review by pvaneck — COMMENTED** [review]
Thanks! Other than a couple minor comments, I think this looks good and is helpful!

File: docs/samples/metrics-and-monitoring/README.md (hunk 1 above)
Comment (shrinandj) [reply docs/samples/metrics-and-monitoring/README.md]: Done

File: docs/samples/metrics-and-monitoring/README.md (hunk 2 above)
Comment (shrinandj) [reply docs/samples/metrics-and-monitoring/README.md:29]: Done

**Comment by yuzisun** [comment]
Thanks @shrinandj !

/lgtm
/approve

**Pull request merged**